### PR TITLE
apiextensions/composite: tame "Successfully selected composition" event

### PR DESCRIPTION
**Pull request by sttts**
### Description of your changes

1. Only emit `Successfully selected composition` event on a change.
2. Add what matters: what was selected: `Successfully selected composition: <name>`.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] ~~Added or updated unit **and** E2E tests for my change.~~
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] ~~Added `backport release-x.y` labels to auto-backport this PR, if necessary.~~
- [ ] ~~Opened a PR updating the [docs], if necessary.~~

[contribution process]: https://git.io/fj2m9
[docs]: https://docs.crossplane.io/contribute/contribute

### CODE REVIEW

File: internal/controller/apiextensions/composite/reconciler.go [hunk 1]
```diff
@@ -512,14 +512,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
...
+	orig := xr.GetCompositionReference()
 	if err := r.composite.SelectComposition(ctx, xr); err != nil {
 		log.Debug(errSelectComp, "error", err)
 		err = errors.Wrap(err, errSelectComp)
 		r.record.Event(xr, event.Warning(reasonResolve, err))
 		xr.SetConditions(xpv1.ReconcileError(err))
 		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, xr), errUpdateStatus)
 	}
-	r.record.Event(xr, event.Normal(reasonResolve, "Successfully selected composition"))
+	if compRef := xr.GetCompositionReference(); (orig == nil) != (compRef == nil) || (orig != nil && *compRef != *orig) {
```
Comment (turkenh) [inline internal/controller/apiextensions/composite/reconciler.go:523]: Wouldn't this crash if `compRef` is `nil` but the `orig` is not? (which may not be possible depending on the implementation of SelectComposition though.

How about the following?

```suggestion
	if compRef := xr.GetCompositionReference(); compRef != nil && (orig == nil || *compRef != *orig) {
```

Comment (sttts) [reply internal/controller/apiextensions/composite/reconciler.go:523]: It won't crash: if orig is also nil, `(orig == nil) != (compRef == nil)` is false, and the second only evaluated `orig != nil`. If orig is not nil, the first clause is true.

Comment (turkenh) [reply internal/controller/apiextensions/composite/reconciler.go:523]: What am I missing? https://go.dev/play/p/A-qNj9pnEs6

Comment (sttts) [reply internal/controller/apiextensions/composite/reconciler.go:523]: Not the if clause is crashing, but the command afterwards.

Comment (sttts) [reply internal/controller/apiextensions/composite/reconciler.go]: We don't need the case of unsetting of the reference. Have switched to your variant. It's easier. Thanks!

**Review by turkenh — APPROVED** [review]
Thanks!

**Review by phisco — APPROVED** [review]

**Pull request merged**